src/event/handlers/CommandParserHandler.ts

```diff
@@ -0,0 +1,27 @@
+import { Constants, Message } from "discord.js";
+import { COMMAND_PREFIX } from "../../config.json";
+import EventHandler from "../../abstracts/EventHandler";
+import CommandFactory from "../../factories/CommandFactory";
+
+class CommandParserHandler extends EventHandler {
+	constructor() {
+		super(Constants.Events.MESSAGE_CREATE);
+	}
+
+	async handle(message: Message): Promise<void> {
+		const commandFactory = new CommandFactory();
+
+		await commandFactory.loadCommands();
```

Comment:
Create & Load the factory in the constructor, saving it to a class property, so that the constructor is only created once, when the `CommandParserHandler` is created.  Then on subsequent message events, it is only utilized, and not created each time.

Reply:
Hmm, just tried to do this, however, because `CommandFactory.loadCommands()` is asynchronous it won't let me put it in the `constructor`. Any ideas?

Reply:
Nevermind, @Jason2605 and I worked it out.